---
title: "week 8"
output: html_document
date: "2025-09-09"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r, chunk1}
suppressPackageStartupMessages({
  library("seqinr") # is a package designed to process and analyse sequence data.
  library("R.utils") # general utilities like zip and unzip
})
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
library("R.utils")

URL="http://ftp.ensemblgenomes.org/pub/bacteria/release-53/fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655_gca_000005845/cds/Escherichia_coli_str_k_12_substr_mg1655_gca_000005845.ASM584v2.cds.all.fa.gz"
download.file(URL,destfile="ecoli_cds.fa.gz")
gunzip("ecoli_cds.fa.gz")
list.files()
```

```{r pressure, echo=FALSE}
length(cds)

```

```{r pressure, echo=FALSE}
head(summary(cds))
```

```{r pressure, echo=FALSE}
unlist(cds)-> dna
dna_composition <- count(dna,1)

barplot(dna_composition,xlab="nucleotides",ylab="frequency", main="E coli CDS composition")
dna_proportion<-dna_composition/sum(dna_composition)
barplot(dna_proportion,xlab="nucleotides",ylab="proportion", main="E coli CDS composition")

grid()
translate(cds[[1]])
prot <- list()
for (i in seq_along(cds)) {prot[[i]] <- translate(cds[[i]]) }
prot <- lapply(cds, translate)

aa <- unique(prot[[2]])
aa <- aa[aa != "*"]
length(aa)
count(prot[[1]],wordsize=1,alphabet=aa)
count(prot[[2]],wordsize=1,alphabet=aa)
uco(cds[[2]])
uco(cds[[2]],index="rscu")
uco(cds[[2]],index="rscu",as.data.frame=TRUE)
prots <- unlist(prot)

mycounts <- count(prots,wordsize=3,alphabet=aa)

str(mycounts)
head(mycounts)
myfreq <- count(prots,wordsize=3,alphabet=aa,freq=TRUE)
sessionInfo()
directory <- "/home/shared/counts"
sampleFiles <- list.files(directory)
sampleCondition <- c('10Days','10Days', '10Days', '45Days','45Days', '45Days')
sampleNames <- gsub("\\.txt$", "", basename(sampleFiles)) 
sampleTable <- data.frame(sampleName = sampleNames,
                          fileName = sampleFiles,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)

sampleTable 
if (!require("BiocManager", quietly = TRUE)) {
install.packages("BiocManager")
}
BiocManager::install("DESeq2")
library("DESeq2")
ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
directory = directory,
design = ~ condition)
```